src/mono/mono/mini/interp/transform.c

```diff
@@ -3939,7 +3940,8 @@ recursively_make_pred_seq_points (TransformData *td, InterpBasicBlock *bb)
 {
 	SeqPoint ** const MONO_SEQ_SEEN_LOOP = (SeqPoint**)GINT_TO_POINTER(-1);
 
-	GArray *predecessors = g_array_new (FALSE, TRUE, sizeof (gpointer));
+	dn_vector_t predecessors = {0,};
+	dn_vector_init_t (&predecessors, SeqPoint*);
```

Comment:
An alternative would be to use dn_vector_ptr_t for this, that will have a little more optimized push_back/pop_back implementations.

Reply:
Can I use a `dn_vector_ptr_t` with `dn_vector_index_t`?  I think the answer is yes....

Reply:
Yes, there is also a dn_vector_ptr_index implemented as:

return dn_vector_index_t ((dn_vector_t *)vector, type, index);

so it will cast to type.

Reply:
Maybe we should add a couple of _t versions to ptr as well? That can be done in dn-vector-ptr.h, like:

dn_vector_ptr_index_t
dn_vector_ptr_at_t
dn_vector_ptr_front_t
dn_vector_ptr_back_t
dn_vector_ptr_data_t

That way you can use the generic dn_vecotr_ptr_t type and still open up to cast to different pointer types when accessing the elements.

Reply:
Added